linkerd/app/outbound/src/discover.rs

```diff
@@ -1,9 +1,10 @@
-use crate::Outbound;
+use crate::{policy, Outbound};
 use linkerd_app_core::{
     profiles,
     svc::{self, stack::Param},
     Error,
 };
+pub use policy::TargetAddr;
```

Comment:
Instead, let's define a `pub struct DiscoverAddr(pub Addr)` and reexport it from the parent so the gateway can provide it is `outbound::DiscoverAddr`